src/jpegxl.imageio/jxloutput.cpp

```diff
@@ -0,0 +1,374 @@
+// Copyright Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: Apache-2.0
+// https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+#include <cassert>
+#include <cstdio>
+#include <vector>
+
+#include <OpenImageIO/filesystem.h>
+#include <OpenImageIO/fmath.h>
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/tiffutils.h>
+
+#include <jxl/encode.h>
+#include <jxl/encode_cxx.h>
+#include <jxl/resizable_parallel_runner_cxx.h>
+
+OIIO_PLUGIN_NAMESPACE_BEGIN
+
+#define DBG if (0)
+
+class JxlOutput final : public ImageOutput {
+public:
+    JxlOutput() { init(); }
+    ~JxlOutput() override { close(); }
+    const char* format_name(void) const override { return "jpegxl"; }
+    int supports(string_view feature) const override
+    {
+        return (feature == "alpha" || feature == "nchannels"
+                || feature == "exif" || feature == "ioproxy"
+                || feature == "tiles");
```

Comment:
I think that since this is buffering all output until the end, it can support "random_access" also because it doesn't care what order you send the scanlines.